### DISCUSSION TITLE
Add support for private go modules on GitLab

### DISCUSSION THREAD

**Discussion by himanshumehta1114** [renovatebot/renovate#10380] [Request Help]
My go module uses some other private go modules from self-hosted GitLab instance. Renovate returns the following error while updating the private modules:

```
 INFO: Repository started (repository=himanshu/go-hello)
       "renovateVersion": "25.38.2"
 INFO: Dependency extraction complete (repository=himanshu/go-hello)
       "baseBranch": "master",
       "stats": {
         "managers": {"gomod": {"fileCount": 1, "depCount": 2}},
         "total": {"fileCount": 1, "depCount": 2}
       }
 INFO: PR updated (repository=himanshu/go-hello, branch=renovate/patch-all-minor-patch)
       "pr": 5,
       "prTitle": "Update module rsc.io/quote to v1.5.2"
 INFO: Branch updated (repository=himanshu/go-hello, branch=renovate/rsc.io-quote-3.x)
       "commitSha": "5c525da"
 WARN: artifactErrors (repository=himanshu/go-hello, branch=renovate/rsc.io-quote-3.x)
       "artifactErrors": [
         {
           "lockFile": "go.sum",
           "stderr": "Command failed: go get -d ./...\ngo: source.company.com/foo/bar@v1.1.3: reading source.company.com/foo/bar/go.mod at revision v1.1.3: unknown revision v1.1.3\n"
         }
       ]
 INFO: Branch updated (repository=himanshu/go-hello, branch=renovate/source.company.com-foo-bar-4.x)
       "commitSha": "049afb8"
 WARN: artifactErrors (repository=himanshu/go-hello, branch=renovate/source.company.com-foo-bar-4.x)
       "artifactErrors": [
         {
           "lockFile": "go.sum",
           "stderr": "Command failed: go get -d ./...\ngo: source.company.com/foo/bar/v4@v4.0.0: reading source.company.com/foo/bar/go.mod at revision v4.0.0: unknown revision v4.0.0\n"
         }
       ]
 INFO: Repository finished (repository=himanshu/go-hello)
       "durationMs": 22921
```

On debugging, I found in [lib/manager/gomod/artifacts.ts](https://github.com/renovatebot/renovate/blob/1b6ecd87c0b9e3e2a872d023857e2ed018c43e7c/lib/manager/gomod/artifacts.ts#L28) that `hostRules` consider `github` platform rules only. Can we add support for other platforms?

To fetch the private modules, Go requires `GOPRIVATE` env variable to be configured. Can we add support for that as well?

**Comment by HonkingGoose** [comment]
> On debugging, I found in [lib/manager/gomod/artifacts.ts](https://github.com/renovatebot/renovate/blob/1b6ecd87c0b9e3e2a872d023857e2ed018c43e7c/lib/manager/gomod/artifacts.ts#L28) that `hostRules` consider `github` platform rules only. Can we add support for other platforms?

I think we already have 2 feature request issues about using GitLab with private go modules:

- https://github.com/renovatebot/renovate/issues/3496
- https://github.com/renovatebot/renovate/issues/4601
